flytekit/common/nodes.py

```diff
@@ -219,8 +272,49 @@ def promote_from_model(cls, model):
         :param flytekit.models.core.workflow.Node model:
         :rtype: SdkNode
         """
-        raise _user_exceptions.FlyteAssertion("An SDK node cannot be instantiated merely from a data model object "
-                                              "because it must be contextualized within a workflow.")
+        id = model.id
+        # This should never be called
+        if id == "start-node" or id == "end-node":
```

Comment:
I think we should embed this check in the workflow code since it's really a workflow concept.

Reply:
it's there as well.